podcasts/Onboarding/Plus/SubscriptionPriceAndOfferView.swift

```diff
@@ -17,64 +17,58 @@ struct SubscriptionPriceAndOfferView: View {
 
     var body: some View {
         VStack(alignment: .leading, spacing: 10) {
-            SubscriptionBadge(tier: product.identifier.subscriptionTier)
-            if let offerDescription = offerDescription(for: product) {
-                Text(offerDescription)
-                    .lineLimit(1)
-                    .fixedSize(horizontal: true, vertical: false)
-                    .foregroundColor(product.identifier.plan == .plus ? Color.black : Color.white)
-                    .padding(EdgeInsets(top: 5, leading: 8, bottom: 5, trailing: 8))
-                    .background(product.identifier.plan == .plus ? Color.plusBackgroundColor2 : Color.patronBackgroundColor)
-                    .textCase(.uppercase)
-                    .font(style: .caption2, weight: .semibold)
-                    .clipShape(.capsule)
-                    .transition(.scale)
+            OfferStack {
+                SubscriptionBadge(tier: product.identifier.subscriptionTier)
+                    .transition(.identity.animation(.none))
+                    .id("sub_badge_" + product.identifier.rawValue)
+
+                if let offerDescription = offerDescription(for: product) {
+                    Text(offerDescription)
+                        .foregroundColor(product.identifier.plan == .plus ? Color.black : Color.white)
+                        .padding(EdgeInsets(top: 5, leading: 8, bottom: 5, trailing: 8))
+                        .background(product.identifier.plan == .plus ? Color.plusBackgroundColor2 : Color.patronBackgroundColor)
+                        .textCase(.uppercase)
+                        .font(style: .caption2, weight: .semibold)
+                        .clipShape(.capsule)
+                }
             }
-            Text(price(for: product))
 
+            Text(price(for: product))
         }
+        .frame(maxWidth: .infinity, alignment: .leading)
         .padding(.bottom, 12)
     }
 
     private func price(for subscriptionInfo: ProductInfo) -> AttributedString {
         let subscriptionPeriod = subscriptionInfo.identifier.productInfo.frequency.description
 
-        guard let offer = subscriptionInfo.offer else {
-            var basePrice =  AttributedString(subscriptionInfo.rawPrice)
-            basePrice.font = .headline
-            basePrice.foregroundColor = mainTextColor
-
-            var basePeriod = AttributedString("/ \(subscriptionPeriod)")
-            basePeriod.foregroundColor = secondaryTextColor
-            basePeriod.font = .footnote
-
-            return basePrice + basePeriod
-        }
-
+        let priceFont = UIFont.font(with: .headline, maxSizeCategory: .extraExtraExtraLarge)
+        let periodFont = UIFont.font(with: .footnote, maxSizeCategory: .extraExtraExtraLarge)
```

Comment:
Using the "dynamically sizing font" here allows us to specify the max size the font can scale to and prevent it from clipping or breaking the UI.